src/transformers/trainer.py

```diff
@@ -2111,6 +2135,8 @@ def _inner_training_loop(
         self._globalstep_last_logged = self.state.global_step
         model.zero_grad()
         grad_norm: Optional[float] = None
+        # LOMO has a slightly different opitmizer API, see: https://github.com/OpenLMLab/LOMO/issues/73#issuecomment-2049612639
+        _is_lomo_optimizer = "Lomo" in self.optimizer.optimizer.__class__.__name__
```

Comment:
Can you check if we can peek at `self.optimizer.optimizer.__module__` maybe? (Or similar). Would be a bit more robust of a check than relying on namings